### DISCUSSION TITLE
Explain that through setting "Easyeffects Sink" as the audio source for an app you can give it raw (unprocessed) audio.

### DISCUSSION THREAD

**Discussion by ghost** [wwmm/easyeffects#2430] [Ideas]
An issue I had is that I followed the guide to get better audio from my laptop's speakers. The issue is that on default audio settings in OBS, it gets the processed audio, and when I play back any video file I recorded, it's as if the effects get applied 2x which destroys the audio quality on my laptop. And perhaps on other devices too (?). I had been googling and searching the easyeffects issues page to find out how to give raw audio to an application until I decided to try using the "Easyeffects Sink" as the desktop audio output and... voila. OBS gets the raw audio output. I also got to know more about this as I checked discussion #1836 where it's explained that applications play audio to the easyeffects sink for then easyeffects to process their audio.

**Comment by wwmm** [comment]
>until I decided to try using the "Easyeffects Sink" as the desktop audio output

Have in mind that if you do that it is totally possible that one day PipeWire starts things in a different order and EasyEffects audio routing will be broken. The proper solution is either putting one of OBS streams in EasyEffects blocklist or using OBS menu to select the correct device.

**Reply by ghost** [reply]
I set it only for OBS, *on obs's audio settings*. It should be fine right? I should've clarified that.

**Reply by wwmm** [reply]
>I set it only for OBS, on obs's audio settings. It should be fine right? I should've clarified that.

Oh... Ok. It is fine if you have selected EasyEffects sink as input source in OBS preferences. Under the hoods OBS will be able to find the sink monitor ports. I thought you had set our sink as the default system output device.

**Reply by ghost** [reply]
Ah no, I hadnt done that. I already know why that's a bad idea.

**Comment by wwmm** [comment]
>how to give raw audio to an application

Oh... In this case you just have to set OBS input device to our virtual sink `monitors`. As even on Pulseaudio sink have monitor ports that act as inputs OBS should allow them to be used.

**Reply by ghost** [reply]
And what would these be? I'm at a loss, I don't know much about pulse/pipewire...

**Reply by wwmm** [reply]
>And what would these be? I'm at a loss, I don't know much about pulse/pipewire...

Monitors are output ports usually intended for level meters. But they are often also useful for audio recorders.

**Comment by wwmm** [comment]
As you configuration seems fine we can close this discussion.
